web_src/js/features/repo-diff.js

```diff
@@ -118,32 +118,24 @@ function onShowMoreFiles() {
   countAndUpdateViewedFiles();
 }
 
-export function initRepoDiffShowMore() {
-  $('#diff-files, #diff-file-boxes').on('click', '#diff-show-more-files, #diff-show-more-files-stats', (e) => {
-    e.preventDefault();
-
-    if ($(e.target).hasClass('disabled')) {
+export function doLoadMoreFiles(link, diffEnd, callback) {
+  const url = `${link}?skip-to=${diffEnd}&file-only=true`;
+  $.ajax({
+    type: 'GET',
+    url,
+  }).done((resp) => {
+    if (!resp) {
+      callback(resp);
       return;
     }
-    $('#diff-show-more-files, #diff-show-more-files-stats').addClass('disabled');
-
-    const url = $('#diff-show-more-files, #diff-show-more-files-stats').data('href');
-    $.ajax({
-      type: 'GET',
-      url,
-    }).done((resp) => {
-      if (!resp) {
-        $('#diff-show-more-files, #diff-show-more-files-stats').removeClass('disabled');
-        return;
-      }
-      $('#diff-too-many-files-stats').remove();
-      $('#diff-files').append($(resp).find('#diff-files li'));
-      $('#diff-incomplete').replaceWith($(resp).find('#diff-file-boxes').children());
-      onShowMoreFiles();
-    }).fail(() => {
-      $('#diff-show-more-files, #diff-show-more-files-stats').removeClass('disabled');
-    });
+    $('body').append($(resp).find('script#diff-data-script'));
```

Comment:
Is it the trick to re-run the scripts? Does it trigger Vue's reactivity?

I am curious whether there are other better solutions. If this is the only solution, it needs enough comments IMO.

Reply:
Yes - by running the script we add elements to the pagedata object. This triggers vue and the filetree and list will render accordingly.

Reply:
i've added a comment explaining this - hope that's what you'd expected